### PR TITLE
Set embedding visual settings before initial render

### PR DESCRIPTION
**Pull request by deniskaber**
Relates https://github.com/metabase/metabase/issues/26988

### Description

This moves embedding settings preparation to happen before the rendering. 
This way we set the state first and then render global styles and content. We avoid extra rerender and possible race conditions with font being applied to a page.

Also we add a code that should dynamically apply embedding settings to the app in "preview mode", which will be used soon in https://github.com/metabase/metabase/issues/35961

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/lib/browser.js [hunk 1]
```diff
@@ -7,7 +7,14 @@ function parseQueryStringOptions(s) {
     if (options[name] === "") {
       options[name] = true;
     } else if (/^(true|false|-?\d+(\.\d+)?)$/.test(options[name])) {
-      options[name] = JSON.parse(options[name]);
+      let parsed;
```
Comment (WiNloSt) [inline frontend/src/metabase/lib/browser.js:10]: Why do we need this change? What problem did you encounter?

File: frontend/src/metabase/app.js [hunk 2]
```diff
@@ -73,6 +73,8 @@ function _init(reducers, getRoutes, callback) {
 
   createTracker(store);
 
+  initializeEmbedding(store);
```
Comment (WiNloSt) [inline frontend/src/metabase/app.js:76]: This change makes sense to me

**Review by WiNloSt — COMMENTED** [review]
I'm not sure about the change in `frontend/src/metabase/lib/browser.js`, otherwise, this PR looks good to me.

**Review by WiNloSt — COMMENTED** [review]
This is almost there 🎉 I'd add a test that actually tests our static embedding preview.

You can add a new test in `e2e/test/scenarios/admin-2/whitelabel.cy.spec.js` and find examples how we visit the static embedding modal from `e2e/test/scenarios/embedding/embedding-questions.cy.spec.js`

File: frontend/src/metabase/lib/dom.js [hunk 3]
```diff
@@ -13,6 +13,13 @@ export const getScrollY = () =>
 // Cypress renders the whole app within an iframe, but we want to exlude it from this check to avoid certain components (like Nav bar) not rendering
 export const isWithinIframe = function () {
   try {
+    if (
+      isCypressActive &&
+      window.location.search.includes("enableCypressIframe=true")
```
Comment (ranquild) [inline frontend/src/metabase/lib/dom.js:18]: Why not test the change using real preview in the app and not with a fake parameter?

File: frontend/src/metabase/lib/embed.js [hunk 4]
```diff
@@ -41,12 +40,24 @@ export function initializeEmbedding(store) {
...
+
+    if (IS_EMBED_PREVIEW) {
+      let currentLocation;
+      store.subscribe(() => {
+        const previousLocation = currentLocation;
+        currentLocation = store.getState().routing.locationBeforeTransitions;
+
+        if (
+          previousLocation &&
+          (currentLocation.search !== previousLocation.search ||
```
Comment (ranquild) [inline frontend/src/metabase/lib/embed.js:52]: It looks dangerous to skip certain updates based on this heuristic. Wdyt about using something more stable? I'm not sure, but maybe even checking for `pathname` is better

**Review by ranquild — COMMENTED** [review]
Comments inline

File: e2e/test/scenarios/sharing/reproductions/26988-embedding-dynamic-settings.cy.spec.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      visitDashboard(card.dashboard_id);
+    });
+
+    cy.icon("share").click();
+    cy.findByRole("button", { name: "Set up" }).click();
+    cy.wait("@dashboard");
+
+    getIframeBody().should("have.css", "font-family", `Lato, sans-serif`);
+
+    cy.findByTestId("embedding-settings").findByText("Font").siblings().click();
```
Comment (WiNloSt) [inline e2e/test/scenarios/sharing/reproductions/26988-embedding-dynamic-settings.cy.spec.js:42]: Let's make this a bit more readable

```suggestion
    cy.findByTestId("embedding-settings").findByLabelText("Font").click();
```

But you have to modify the markup a bit as well.

Here's what I did to the [`DisplayOptionSection`](https://github.com/metabase/metabase/blob/6fbc94fe67463ec292b995416c5eb602659dcb93/frontend/src/metabase/public/components/widgets/DisplayOptionsPane.jsx#L129-L134), but maybe you could come up with something better.

```ts
const DisplayOptionSection = ({ title, children }) => {
  const uniqueId = useUniqueId("display-option");
  return (
    <DisplayOption>
      <DisplayOptionTitle id={uniqueId}>{title}</DisplayOptionTitle>
      <span aria-labelledby={uniqueId}>{children}</span>
    </DisplayOption>
  );
};
```

**Review by WiNloSt — APPROVED** [review]
Looks good to me, but if you could fix the E2E before merging that'd be great.

**Pull request merged**